raw-svo.setup.lua

```diff
@@ -256,6 +259,101 @@ signals.gmcpiretimeupdate:connect(function()
   end
 end)
 
+signals.gmcpcharafflictionslist = luanotify.signal.new()
+signals.gmcpcharafflictionsremove = luanotify.signal.new()
+signals.gmcpcharafflictionsadd = luanotify.signal.new()
+
+signals.gmcpchardefenceslist = luanotify.signal.new()
+signals.gmcpchardefencesremove = luanotify.signal.new()
+signals.gmcpchardefencesadd = luanotify.signal.new()
+
+
+signals.gmcpcharafflictionsadd:connect(function()
+  local thisaff = gmcp.Char.Afflictions.Add.name
+  if thisaff:sub(-4) == " (1)" then thisaff = thisaff:sub(1, -5) end
+  gaffl[thisaff] = true
+  if conf.gmcpaffechoes then echof("Gained aff %s", thisaff) end
+  if dict.sstosvoa[thisaff] then
+    addaff(dict[dict.sstosvoa[thisaff]])
+  end
+end)
+
+signals.gmcpcharafflictionsremove:connect(function()
+  local thisaff = gmcp.Char.Afflictions.Remove[1]
+  gaffl[thisaff] = nil
+  if conf.gmcpdefechoes then echof("Cured aff %s", thisaff) end
+  if dict.sstosvoa[thisaff] then
+    removeaff(dict.sstosvoa[thisaff])
+  end
+end)
+
+signals.gmcpcharafflictionslist:connect(function()
+  gaffl = {}
+  local preaffl = {}
+  for key, val in ipairs(affl) do preaffl[val] = true end
+  
+  for index, val in ipairs(gmcp.Char.Afflictions.List) do
+    local thisaff = val.name
+    if thisaff:sub(-4) == " (1)" then thisaff = thisaff:sub(1, -5) end  
+    gaffl[thisaff] = true
+    if preaffl[thisaff] then
+      preaffl[thisaff] = false
+    elseif dict.sstosvoa[thisaff] then
+      addaff(dict[dict.sstosvoa[thisaff]])
+    end
+  end
+  
+  for key, val in pairs(preaffl) do
+    if val and dict.svotossa[thisaff] then removeaff(key) end
+  end
+end)
+
+
+signals.gmcpchardefencesadd:connect(function()
+  thisdef = gmcp.Char.Defences.Add.name
+  gdefc[thisdef] = true
+  if conf.gmcpdefechoes then echof("Gained def "..thisdef) end
+  if dict.sstosvod[thisdef] then
+    defs["got_"..dict.sstosvod[thisdef]]()
+  end
+end)
+
+signals.gmcpchardefencesremove:connect(function()
+  thisdef = gmcp.Char.Defences.Remove[1]
+  gdefc[thisdef] = nil
+  if conf.gmcpdefechoes then echof("Lost def "..thisdef) end
+  if dict.sstosvod[thisdef] then
+    defs["lost_"..dict.sstosvod[thisdef]]()
+  end
+end)
+
+signals.gmcpchardefenceslist:connect(function()
+  gdefc = {}
+  local predefs = table.deepcopy(defc)
```

Comment:
careful about `table.deepcopy`. Mudlet 2.1 doesn't have that yet, but svof supplies a local `deepcopy` function.